src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -471,3 +471,55 @@ lemma ideal.homogeneous_hull_eq_Inf (I : ideal A) :
 eq.symm $ is_glb.Inf_eq $ (ideal.homogeneous_hull.gc 𝒜).is_least_l.is_glb
 
 end galois_connection
+
+section irrelavent_ideal
+
+open graded_algebra finset set_like.graded_monoid
+
+variables [comm_semiring R] [semiring A]
+variables [algebra R A] [decidable_eq ι] [add_monoid ι]
+variables (𝒜 : ℕ → submodule R A) [graded_algebra 𝒜]
```

Comment:
Does `canonically_ordered_add_monoid ι` generalize this?

Reply:
Yes, I think so. It can also be generalised to sets with negative part by taking irrelevant ideal as `{a | ∀ (i : ι), i ≤ 0 → proj 𝒜 i a = 0}`. But I don't know if it is useful.

I only intend to use this in `Proj` construction where I only used ℕ-grading

Reply:
I think `canonically_ordered_add_monoid` is enough for now, since that shouldn't actually require the proof to change in any meaningful way. If you like, maybe add a comment explaining that what you suggest could be a future generalization.

Reply:
I have updated it to `canonically_ordered_add_monoid`. Thank you for the advice. Do you think it is worth it to define it as kernel of `proj 𝒜 ⊥`

Reply:
Yes, I think defining it that way would be a neat way to handle it, although probably as with 0 not bot (they're equal anyway)

Reply:
In a genreal `canonically_ordered_add_monoid`, bot is not always equal to zero, right?
For example consider `{-1, 0, 1, 2, 3, ....}`. Then `proj 𝒜 ⊥` is not a ring homomorphism because `1 : A` is not in `𝒜 ⊥`; similarly `proj 𝒜 0` is not a ring homomorphism, because `proj 𝒜 0 (x + y)` may not equal to `proj 𝒜 0 x + proj 𝒜 0 y`. So for `proj 𝒜 0` to be ring hom, we need `0 = bot`. Then canonically ordered add monoid is (almost?) just Nat.

Reply:
It is: https://leanprover-community.github.io/mathlib_docs/algebra/order/monoid.html#bot_eq_zero

Reply:
Thank you!